pkg/cache/cache.go

```diff
@@ -0,0 +1,39 @@
+package cache
+
+import (
+	"net/http/httputil"
+	"sync"
+)
+
+type ProxiesMap = map[string]*httputil.ReverseProxy
+
+type CacheManager struct {
+	proxiesMap *ProxiesMap
+	mutex      *sync.Mutex
+}
+
+func NewCacheManager() *CacheManager {
+	return &CacheManager{
+		proxiesMap: &ProxiesMap{},
+		mutex:      &sync.Mutex{}}
+}
+
+func (manager *CacheManager) GetProxy(proxiesMapKey string) *httputil.ReverseProxy {
+	manager.mutex.Lock()
+	defer func() {
+		manager.mutex.Unlock()
+	}()
+	return (*manager.proxiesMap)[proxiesMapKey]
+}
+
+func (manager *CacheManager) SetProxy(proxiesMapKey string, proxy *httputil.ReverseProxy) {
+	manager.mutex.Lock()
+	(*manager.proxiesMap)[proxiesMapKey] = proxy
+	manager.mutex.Unlock()
+}
+
+func (manager *CacheManager) Delete(proxiesMapKey string) {
```

Comment:
This is unused. We could leave it we want to plan to add some cache invalidation later, but otherwise it can get removed

Reply:
I removed it in the latest push.